src/layout/layout.rs

```diff
@@ -812,53 +961,164 @@ impl Layout {
+        let changes: HashMap<Variable, f64> = solver.fetch_changes().iter().copied().collect();
+
+        // please leave this comment here as it's useful for debugging unit tests when we make any
+        // changes to layout code - we should replace this with tracing in the future.
+        // let ends = format!(
+        //     "{:?}",
+        //     segments
+        //         .iter()
+        //         .map(|e| changes.get(&e.end).unwrap_or(&0.0))
+        //         .collect::<Vec<&f64>>()
+        // );
+        // dbg!(ends);
+
+        let inner = area.inner(&layout.margin);
+        let segment_rects = Layout::get_results(&changes, inner, layout, &segments);
+        let spacer_rects = Layout::get_results(&changes, inner, layout, &spacers);
+
+        Ok((segment_rects, spacer_rects))
+    }
+
+    /// Wrapper function around the cassowary-rs solver to be able to split a given area into
+    /// smaller ones based on the preferred widths or heights and the direction.
+    ///
+    /// Note that the constraints are applied to the whole area that is to be split, so using
+    /// percentages and ratios with the other constraints may not have the desired effect of
+    /// splitting the area up. (e.g. splitting 100 into [min 20, 50%, 50%], may not result in [20,
+    /// 40, 40] but rather an indeterminate result between [20, 50, 30] and [20, 30, 50]).
+    ///
+    /// This method stores the result of the computation in a thread-local cache keyed on the layout
+    /// and area, so that subsequent calls with the same parameters are faster. The cache is a
+    /// LruCache, and grows until [`Self::DEFAULT_CACHE_SIZE`] is reached by default, if the cache
+    /// is initialized with the [Layout::init_cache()] grows until the initialized cache size.
+    ///
+    /// There is a helper method on Rect that can be used to split the whole area into smaller ones
+    /// based on the layout: [`Rect::split()`]. That method is a shortcut for calling this method.
+    /// It allows you to destructure the result directly into variables, which is useful when you
+    /// know at compile time the number of areas that will be created.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// # use ratatui::prelude::*;
+    /// let layout = Layout::default()
+    ///     .direction(Direction::Vertical)
+    ///     .constraints([Constraint::Length(5), Constraint::Min(0)])
+    ///     .split(Rect::new(2, 2, 10, 10));
+    /// assert_eq!(layout[..], [Rect::new(2, 2, 10, 5), Rect::new(2, 7, 10, 5)]);
+    ///
+    /// let layout = Layout::default()
+    ///     .direction(Direction::Horizontal)
+    ///     .constraints([Constraint::Ratio(1, 3), Constraint::Ratio(2, 3)])
+    ///     .split(Rect::new(0, 0, 9, 2));
+    /// assert_eq!(layout[..], [Rect::new(0, 0, 3, 2), Rect::new(3, 0, 6, 2)]);
+    /// ```
+    pub fn split(&self, area: Rect) -> Rects {
+        self.split_with_spacers(area).0
+    }
```

Comment:
`split` returns just the first element in the tuple, which contains the same results as before.